app/src/main/java/com/babylon/wallet/android/presentation/navigation/NavigationHost.kt

```diff
@@ -139,12 +140,14 @@ fun NavigationHost(
             }
         )
         addSingleMnemonic(
-            navController = navController,
             onBackClick = {
                 navController.popBackStack()
             },
-            onStartRecovery = {
```

Comment:
are you passing mnemonic via compose args to remove shared view model? Are we sure we want to do this? I believe at the time of developing this feature android team discouraged passing mnemonic via args, as it might be saved out of our control on the navstack for purpose of android framework restoring app state. I didn't dig deeper into it, let me know what you think

Reply:
True, and thanks for bringing this up. It was in my mind to discuss about it in the android chapter meeting.

Anyway, I did a research on this and couldn't find any evidence of potential dangers. And I would really like your insights on this. Based on this and on the fact that these args will be passed only when user recover from onboarding which means, no network, no profile is yet initialized, and biometric auth is needed in the end of the flow I decided to pass them through nav args.

Please let me know if you have more info for this, and I am fine to change it back to shared view model or any other way. 
(Tbh I also **feel** that this is not safe, I just can't prove it 😅  )

Reply:
if security risk of both solutions would be equal, this change here is excellent improvement for overall logic readability though